src/com/cdd/bao/template/SchemaVocab.java

```diff
@@ -359,6 +391,9 @@ public void debugSummary()
 			Util.writeln("    # nodes: " + stored.tree.getFlat().length + ", depths:" + str);
 		}
 	}
+
+	// get remappings
+	public Map<String, StoredRemapTo> getRemappings() {return Collections.unmodifiableMap(remappings);}
```

Comment:
I didn't know about `Collections.unmodifiableMap`... this is good stuff. I think it can & should be used in other places too, so by all means upgrade them when you stumble into them.